bob-access/src/error.rs

```diff
@@ -0,0 +1,61 @@
+use axum::{
+    response::{IntoResponse, Response},
+    Json,
+};
+use http::StatusCode;
+use serde_json::json;
+use std::{error::Error as StdError, fmt::Display};
+use tonic::codegen::http::header::ToStrError;
+
+#[derive(Debug)]
+pub enum Error {
+    _Unknown,
+    InvalidToken(String),
+    Validation(String),
+    Os(String),
+    UserNotFound,
+    ConversionError(ToStrError),
+    CredentialsNotProvided(String),
+    MultipleCredentialsTypes,
+    UnauthorizedRequest,
+    PermissionDenied,
+    NotGrpcRequest,
+}
+
+impl Error {
+    pub fn description(&self) -> (StatusCode, &'static str) {
+        use Error::*;
+        match self {
+            _Unknown => (StatusCode::INTERNAL_SERVER_ERROR, "Unknown error"),
+            InvalidToken(_) => (StatusCode::BAD_REQUEST, "Invalid token"),
+            Validation(_) => (StatusCode::BAD_REQUEST, "Validation error"),
+            Os(_) => (StatusCode::INTERNAL_SERVER_ERROR, "Os error"),
+            UserNotFound => (StatusCode::BAD_REQUEST, "User not found"),
+            ConversionError(_) => (StatusCode::BAD_REQUEST, "Conversion error"),
+            CredentialsNotProvided(_) => (StatusCode::BAD_REQUEST, "Credentials not provided"),
```

Comment:
This should be a `FORBIDDEN` status code

Reply:
@ikopylov You previously suggested to use UNAUTHORIZED

Reply:
Sorry. 
According to [this](https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses) the best status code is `UNAUTHORIZED` (which actually means `not authenticated`). `FORBIDDEN` is used when the user does not have permission to call the method.